src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -145,6 +145,13 @@ try
     {
         // TODO: "relative" bold?
         weight = DWRITE_FONT_WEIGHT_BOLD;
+        // Since we are setting the font weight according to the text attribute,
+        // make sure to tell the text format to ignore the user set font weight
+        _fontRenderData->UseUserWeight(false);

```

Comment:
So this is basically `_fontRenderData->UseUserWeight(!drawingContext->useBoldFont);`?

Would it be less confusing if we flipped the meaning of either of the two booleans?

Reply:
> So this is basically `_fontRenderData->UseUserWeight(!drawingContext->useBoldFont);`?

Yes, but I did not want to access `drawingContext->useBoldFont` again since we did that in the `if` statement to get in this block in the first place

Reply:
Yeah that makes sense.

But regarding my other point: Would it help the code clarity if we kept both booleans in sync?
If we had something like `InhibitUserWeight()` we'd knew that it's always true if `drawingContext->useBoldFont` is true. I don't know about others, but I'd personally would find it easier to understand the state-logic of the renderer that way.

Reply:
I don't feel strongly either way but its a small enough fix, made the change!